buildpacks/python/python/validation.go

```diff
@@ -13,117 +12,92 @@ import (
 	"github.com/buildpacks/libcnb"
 	"github.com/paketo-buildpacks/libpak"
 	"github.com/paketo-buildpacks/libpak/bard"
+
+	"kn-fn/shared/shared"
 )
 
-type FunctionValidationLayer struct {
+type FunctionValidator struct {
 	layerContributor libpak.LayerContributor
+	commandRunner    shared.CommandRunner
 	logger           bard.Logger
 
+	applicationPath string
 	module          string
 	function        string
-	applicationPath string
 
-	override bool
-
-	Invoker    Layer
-	InvokerDep Layer
+	invoker     Layer
+	invokerDeps Layer
 }
 
-type FunctionValidationOpts func(*FunctionValidationLayer, map[string]string)
+type FunctionValidatorOpts func(*FunctionValidator, map[string]string)
 
+//go:generate mockgen -destination ../mock_python/layer.go . Layer
 type Layer interface {
 	PythonPath() string
 }
 
-func NewFunctionValidationLayer(appPath string, invoker Layer, InvokerDepLayer Layer, opts ...FunctionValidationOpts) FunctionValidationLayer {
-	fvl := FunctionValidationLayer{
+func NewFunctionValidator(
+	appPath string,
+	functionClass string,
+	invoker Layer,
+	InvokerDepLayer Layer,
+	commandRunner shared.CommandRunner,
+	logger bard.Logger,
+) *FunctionValidator {
+	parts := strings.Split(functionClass, ".")
+
+	fv := &FunctionValidator{
 		applicationPath: appPath,
-		Invoker:         invoker,
-		InvokerDep:      InvokerDepLayer,
+		module:          parts[0],
+		function:        parts[1],
+		invoker:         invoker,
+		invokerDeps:     InvokerDepLayer,
+		commandRunner:   commandRunner,
+		logger:          logger,
 	}
 	meta := map[string]string{}
 
-	for _, opt := range opts {
-		opt(&fvl, meta)
-	}
-
-	fvl.layerContributor = libpak.NewLayerContributor("validation", meta, libcnb.LayerTypes{
+	fv.layerContributor = libpak.NewLayerContributor("validation", meta, libcnb.LayerTypes{
+		// NOTE: need uncached build layer so validation runs for rebuilds
 		Build: true,
+		Cache: false,
 	})
 
-	return fvl
+	return fv
 }
 
-func (i FunctionValidationLayer) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
-	i.layerContributor.Logger = i.logger
+func (fv *FunctionValidator) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
```

Comment:
Just wanted to call out this is a good refactor `i` --> `fv`, but you could definitely make a PR of _just_ refactors in the future to avoid a multi-hundred (or even multi-thousand) line PR and get this type of change in sooner.